gql_schema_codegen/dependency/dependency.py

```diff
@@ -1,12 +1,83 @@
 from itertools import groupby
-from typing import List, NamedTuple, Set
+from typing import Dict, List, NamedTuple, Set
 
 
 class Dependency(NamedTuple):
     imported_from: str
     dependency: str
 
 
+INTERMEDIATE_INTERFACES: Dict[str, str] = {}
+
+
+def get_interface_dependencies():
+    return INTERMEDIATE_INTERFACES
+
+
+def update_interface_dependencies(config_file_content):
+    global INTERMEDIATE_INTERFACES
+    if type(config_file_content) is dict:
+        data = config_file_content.get("interfaceInheritance")
+        if type(data) is dict:
+            INTERMEDIATE_INTERFACES = data
+
+
+def remove_interface_dependencies(
+    interfaces: List[str],
+    intermediate_interfaces: Dict[str, str] = {},
+) -> List[str]:
+    """Filter all dependencies from intermediate interfaces
+
+    Assumes that all keys in intermediate_interfaces are leaf nodes and
+    returns all other parent interfaces from a given list
+
+    Intemediate interfaces in GraphQL implement other interfaces. This is part
+    of the spec (see https://github.com/graphql/graphql-spec/pull/373)
+    but not implemented in all clients yet (e.g., neo4j)
+    Thus we are parsing intermediate interfaces in scalars.yml so as to emit
+    proper python code without relying on the graphql schema being correct
+
+
+    >>> remove_interface_dependencies(['t1', 't2', 'i1', 'i2', 't3'], {'i1':'i2'})
+    ['t1', 't2', 'i1', 't3']
+
+    >>> remove_interface_dependencies(['t1', 't2', 't3'], {})
+    ['t1', 't2', 't3']
+
+    >>> remove_interface_dependencies(['t1', 't2', 'i1', 'i2', 'i3', 't3'], \
+                                      {'i1':'i2', 'i2': 'i3'})
+    ['t1', 't2', 'i1', 't3']
+
+    >>> remove_interface_dependencies(['t1', 't2', 'i1', 'i2', 'i3', \
+                                       'ni1', 'ni2', 't3'], \
+                                      {'i1':'i2', 'i2': 'i3', 'ni1':'ni2'})
+    ['t1', 't2', 'i1', 'ni1', 't3']
+    """
+    deps: Set[str] = set()
+    if not intermediate_interfaces:
+        intermediate_interfaces = INTERMEDIATE_INTERFACES
```

Comment:
```suggestion
    intermediate_interfaces = intermediate_interfaces or INTERMEDIATE_INTERFACES
```

usually simplifies things a bit, especially if you count the mccabe function complexity its nice to reduce any direct if statements to more simple alternatives